satpy/utils.py

```diff
@@ -631,6 +633,112 @@ def _get_pytroll_chunk_size():
+    This is done by multiplying the input array shape by the
+    ``low_res_multiplier`` and then using Dask's utility functions and
+    configuration to produce a chunk size to fit into a specific number of
+    bytes. See :ref:`dask:automatic-chunking` for more information.
+    Next, the same multiplier is used to reduce the high resolution chunk sizes
+    to the lower resolution of the input data. The end result of reading
+    multiple resolutions of data is that each dask chunk covers the same
+    geographic region. This also means replicating or aggregating one
+    resolution and then combining arrays should not require any rechunking.
+
+    .. note::
+
+        Only 2 or 3-dimensional shapes are supported. In the case of 3D arrays
+        the first dimension is assumed to be "bands" and is given a chunk
+        size of 1. For shapes with other numbers of dimensions, the chunk size
+        for the entire array is determined by dask's "auto" chunking and
+        resolution is ignored.
+
+    Args:
+        input_shape: Shape of the array to compute dask chunk size for.
+        input_dtype: Dtype for the final unscaled array. This is usually
+            32-bit float (``np.float32``) or 64-bit float (``np.float64``)
+            for non-category data. If this doesn't represent the final data
+            type of the data then the final size of chunks in memory will not
+            match the user's request via dask's ``array.chunk-size``
+            configuration. Sometimes it is useful to keep this as a single
+            dtype for all reading functionality (ex. ``np.float32``) in order
+            to keep all read variable chunks the same size regardless of dtype.
+        num_high_res_elements: Smallest number of high (fine) resolution
+            elements that make up a single "unit" or chunk of data. This could
+            be a multiple or factor of the scan size for some instruments and/or
+            could be based on the on-disk chunk size. This value ensures that
+            chunks are aligned to the underlying data structure for best
+            performance.
+        low_res_multiplier: Number of high (fine) resolution pixels that fit
+            in a single low (coarse) resolution pixel.
+        whole_scan_width: To create the entire width (x dimension) of the
+            array as a single chunk. This is useful in cases when future
+            operations will operate on entire instrument scans of data at
+            a time. For example, polar-orbiter scan geolocation being
+            interpolated from low resolution to high resolution.
+
+    Returns:
+        A tuple where each element is the chunk size for that axis/dimension.
+
+    """
+    if len(input_shape) not in (2, 3):
+        # we're not sure about this shape so don't guess
+        return dask.array.core.normalize_chunks("auto", shape=input_shape, dtype=input_dtype)
```

Comment:
Codecov says this is not tested?

Reply:
Codecov is wrong. It is covered when run locally. I'm not sure why coveralls isn't being reported in the PR. I had to make a small change to fix the docs build so let's see if it recognizes the coverage with this new run.